TTS/tts/models/base_tts.py

```diff
@@ -416,7 +416,10 @@ def on_init_start(self, trainer):
         """Save the speaker.pth and language_ids.json at the beginning of the training. Also update both paths."""
         if self.speaker_manager is not None:
             output_path = os.path.join(trainer.output_path, "speakers.pth")
-            self.speaker_manager.save_ids_to_file(output_path)
+            if trainer.config.use_d_vector_file and trainer.config.d_vector_file:
+                self.speaker_manager.save_embeddings_to_file(output_path)
+            else:
+                self.speaker_manager.save_ids_to_file(output_path)
```

Comment:
In the past we was saving the speaker's d_vectors for each run however it is not good because depending on your dataset the speaker's d_vectors file will be too large and it can cause problems with disk space. The "speakers_file" and  "speaker_file" conditionals that changed in "[TTS/tts/utils/speakers.py](https://github.com/coqui-ai/TTS/pull/2234/files#diff-783c5c4b71a0948639ef6a72f085ef1e4a8c1c36fda356b9510de7512cf549fe)" was kept just to maintain compatibility with old versions.

 The real problem is the Type of d_vector_file inside of VitsArgs and VitsConfig. It should be "List[str]" and not "str".

Reply:
Okay this has become a very serious task, I think won't be able to do it as I started with the codebase 3 days ago :D
The whole codebase uses d_vector_file as a single file
like in 
/workspaces/TTS/TTS/tts/configs/align_tts_config.py
/workspaces/TTS/TTS/tts/utils/speakers.py
/workspaces/TTS/TTS/tts/configs/fast_pitch_config.py
glow_tts, overflow_tts, etc...

I think someone with better code understanding can do it!

Reply:
Thanks. Also can you please look at this https://github.com/coqui-ai/TTS/issues/2236#issuecomment-1362804774